src/state/opamSysPoll.ml

```diff
@@ -98,13 +98,17 @@ let poll_os_distribution () =
     (if is_android () then Some "android" else
      os_release_field "ID" >>= norm >>+ fun () ->
      command_output ["lsb_release"; "-i"; "-s"] >>= norm >>+ fun () ->
-     try
-       List.find Sys.file_exists ["/etc/redhat-release";
-                                  "/etc/centos-release";
-                                  "/etc/gentoo-release";
-                                  "/etc/issue"] |>
-       fun s -> Scanf.sscanf s " %s " norm
-     with Not_found -> linux)
+     let release_file = List.find_opt Sys.file_exists ["/etc/redhat-release";
+                                                       "/etc/centos-release";
+                                                       "/etc/gentoo-release";
+                                                       "/etc/issue"]
+     in
+     match release_file with
+     | None -> linux
+     | Some release_file -> (
+       match OpamProcess.read_lines release_file with
+       | [] -> linux
+       | s::_ -> Scanf.sscanf s " %s " norm))
```

Comment:
Proposal, some nitpicking: indentation, readabiliy & catch `Scanf.sscanf` failure

```suggestion
     let release_file =
       List.find_opt Sys.file_exists ["/etc/redhat-release";
                                      "/etc/centos-release";
                                      "/etc/gentoo-release";
                                      "/etc/issue"]
     in
     match OpamStd.Option.map OpamProcess.read_lines release_file with
     | None |  Some [] -> linux
     | Some (s::_) ->
       try Scanf.sscanf s " %s " norm
       with Scanf.Scan_failure _ -> linux)
```

Reply:
Yep, that's nicer indeed, thanks!